$NetBSD: patch-meson.build,v 1.1 2025/10/31 12:29:12 jperkin Exp $

Disable libsocket handling on SunOS.  meson cannot handle the fact that this
library is not provided by pkg-config and refuses to build gdk-pixbuf-loader.
We already just link against it directly anyway.

IRIX: ask rustc for the native libraries the Rust code needs, as on Linux
(-lunwind -lpthread -lm -lc): the C link of librsvg-2.so otherwise misses the
unwinder (undefined _Unwind_Resume), which IRIX's C driver does not add.

--- meson.build.orig
+++ meson.build
@@ -180,7 +180,7 @@
 # into '-lsocket' and '-lnsl'.  Anything that calls socketpair(), getifaddr(),
 # etc. probably needs to include network_deps
 #
-if host_system == 'sunos'
+if host_system == 'meson-dependencies-are-broken'
   network_deps = [
     cc.find_library('socket', required: false),
   ]
@@ -340,7 +340,7 @@
 endif
 
 default_overrides = []
-if host_system in ['windows', 'linux']
+if host_system in ['windows', 'linux', 'irix']
   rustc_query_native_static_libs_args = rustc_query_args
   if target_arg != []
     rustc_query_native_static_libs_args += target_arg
